File: Medium leetcode/3271. Hash Divided String.java
class Solution
{
    public String stringHash(String s, int k)
    {
        StringBuilder ans = new StringBuilder();
        for (int i = 0; i < s.length(); i += k)
        {
            int sum = 0;
            for (int j = 0; j < k && i + j < s.length(); j++)
            {
                sum += s.charAt(i + j) - 'a';
            }
            ans.append((char)(sum % 26 + 'a'));
        }
        return ans.toString();
    }
}

File: Medium leetcode/3271. Hash Divided String.py
class Solution:
    def stringHash(self, s: str, k: int) -> str:
        ans = ""
        for i in range(0, len(s), k):
            sum = 0
            for j in range(k):
                if i + j < len(s):
                    sum += ord(s[i + j]) - ord('a')
            ans += chr((sum % 26) + ord('a'))
        return ans

